core/test/solver/gmres.cpp

```diff
@@ -101,165 +109,193 @@ class Gmres : public ::testing::Test {
 };
 
 
-TEST_F(Gmres, GmresFactoryKnowsItsExecutor)
+template <typename T>
+constexpr gko::remove_complex<T> Gmres<T>::reduction_factor;
```

Comment:
Yes it is needed, because it's a static variable declared in the class and static members always need to have a definition outside of the class to be complete.

See: https://en.cppreference.com/w/cpp/language/static

Reply:
Thanks for explanation.